src/moonlink_service/src/rest_api.rs

```diff
@@ -161,7 +162,31 @@ pub struct HealthResponse {
     pub timestamp: u64,
 }
 
-/// Create the router with all API endpoints
+/// Map backend error to appropriate HTTP status code based on error type
+fn get_backend_error_status_code(error: &moonlink_backend::Error) -> StatusCode {
+    match error {
+        moonlink_backend::Error::InvalidArgumentError(_)
+        | moonlink_backend::Error::ParseIntError(_)
+        | moonlink_backend::Error::Json(_) => StatusCode::BAD_REQUEST,
+
+        _ => match error.get_status() {
+            ErrorStatus::Temporary => StatusCode::SERVICE_UNAVAILABLE,
+            ErrorStatus::Permanent => StatusCode::INTERNAL_SERVER_ERROR,
+        },
+    }
+}
+
+/// Map specific error contexts that don't have backend errors
+fn get_context_error_status_code(error_type: &str) -> StatusCode {
```

Comment:
I'm not sure how useful the util function is, what about we directly assigning status code?